src/app/pages/client-review/review.component.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { NgxSpinnerModule } from 'ngx-spinner';
+import { ReactiveFormsModule, FormsModule, NgForm } from '@angular/forms';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ToastrModule } from 'ngx-toastr';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { throwError, of, Subject } from 'rxjs';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { ReviewComponent } from './review.component';
+
+class RouterStub {
+  navigate(params) {
+  }
+}
+class ActivatedRouterStub {
+  private subject = new Subject();
+
+  push(value) {
+    this.subject.next(value);
+  }
+  get params() {
+    return this.subject.asObservable();
+  }
+}
+
+describe('ReviewComponent', () => {
+  let component: ReviewComponent;
+  let fixture: ComponentFixture<ReviewComponent>;
+  const mockRouter = jasmine.createSpyObj(['navigate']);
+  const mockToastr = jasmine.createSpyObj(['error']);
+  const mockSpinner = jasmine.createSpyObj(['show', 'hide']);
```

Comment:
@Niyitangasam , I believe we have spies for this functionality already setup in the `spies.ts` file

Reply:
@Tevinthuku , I worked on this.